src/main/java/ui/export/PDFCompiler.java

```diff
@@ -0,0 +1,103 @@
+/* Licensed under MIT 2024. */
+package ui.export;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.interactive.form.PDAcroForm;
+import ui.Time;
+import ui.json.Global;
+import ui.json.JSONHandler;
+import ui.json.Month;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+
+public class PDFCompiler {
+
+	private PDFCompiler() {
+		throw new IllegalAccessError();
+	}
+
+	public static Optional<String> compileToPDF(Global global, Month month, File targetFile) {
+		try (InputStream templateStream = Object.class.getResourceAsStream("/pdf/template.pdf")) {
+			if (templateStream == null) {
+				return Optional.of("Template PDF not found in resources.");
+			}
+
+			PDDocument document = PDDocument.load(templateStream);
+			return writeToPDF(document, global, month, targetFile);
+
+		} catch (IOException e) {
+			return Optional.of(e.getMessage());
+		}
+	}
+
+	private static Optional<String> writeToPDF(PDDocument document, Global global, Month month, File targetFile) throws IOException {
+		PDAcroForm form = document.getDocumentCatalog().getAcroForm();
+		if (form == null) {
+			return Optional.of("No form found in the document. Nothing we can do, sorry.");
+		}
+
+		form.getField("GF").setValue(global.getNameFormalFormat()); // Name
+		form.getField("abc").setValue(String.valueOf(month.getMonth())); // Month
+		form.getField("abdd").setValue(String.valueOf(month.getYear())); // Year
```

Comment:
Could you use proper placeholders as for the other values :)

Reply:
No, these are the names of the fields in the PDF. To get all the names, I iterated through all the fields and, if possible, just filled it with its identifier, so that's how I got them.
These are the names from the official template from the KIT Website that I downloaded.